src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -288,9 +289,49 @@ private async Task PublishTestRunDataAsync(String teamProject, TestRunContext te
                 _executionContext.Error(StringUtil.Loc("FailedTestsInResults"));
             }
 
-            await PublishEventsAsync(connection);
+            await PublishEventsAsync(connection); 
+            TriggerCoverageMergeJob(_testResultFiles, _executionContext);      
         }
-
+        
+        // Queue code coverage merge job if code coverage attachments are published to avoid BQC timeout.
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "connection")]
+        private void TriggerCoverageMergeJob(List<string> resultFilesInput , IExecutionContext context)
```

Comment:
Please wrap this method in `try-catch`, so that it doesn't affect the main flow.